packages/block-library/src/comment-content/index.php

```diff
@@ -28,6 +28,18 @@ function render_block_core_comment_content( $attributes, $content, $block ) {
 		return '';
 	}
 
+	$moderation_note = '';
+	if ( '0' === $comment->comment_approved ) {
+		$commenter = wp_get_current_commenter();
+
+		if ( $commenter['comment_author_email'] ) {
+			$moderation_note = __( 'Your comment is awaiting moderation.' );
+		} else {
+			$moderation_note = __( 'Your comment is awaiting moderation. This is a preview; your comment will be visible after it has been approved.' );
```

Comment:
```suggestion
			$moderation_note = __( 'Your comment is awaiting moderation. This is a preview; your comment will be visible after it has been approved.' );
			// Using moderation-hash, strip HTML tags from comment.
			$comment_text = wp_kses( $comment_text, array() );
```

Per note on the original issue about stripping HTML in the preview. Spammers were exploiting including links in unmoderated comments.

Reply:
I added the same validation as in Core. Allowing links if the user has the Cookie setting enabled.

![Screenshot 2022-04-27 at 16 49 09](https://user-images.githubusercontent.com/37012961/165546292-8e07903a-1d26-4b46-aa93-cbab06967c7b.png)

I will create another bug fix for the Author link.